src/compton/test/t2Compton.cc

```diff
@@ -0,0 +1,322 @@
+//-----------------------------------*-C++-*----------------------------------//
+/*!
+ * \file   compton/test/t2Compton.cc
+ * \author Andrew Till
+ * \date   2020 Oct 14
+ * \brief  Implementation file for t2Compton
+ * \note   Copyright (C) 2017-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//------------------------------------------------------------------------------------------------//
+
+#include "c4/ParallelUnitTest.hh"
+#include "compton/Compton.hh"
+#include "ds++/Release.hh"
+#include "ds++/Soft_Equivalence.hh"
+#include <fstream>
+#include <iomanip>
+#include <sstream>
+#include <stdexcept>
+#include <vector>
+
+namespace rtt_compton_test {
+
+using rtt_dsxx::soft_equiv;
+
+//------------------------------------------------------------------------------------------------//
+// TESTS
+//------------------------------------------------------------------------------------------------//
+
+//!  Tests the Compton constructor and a couple of access routines.
+void compton_file_test(rtt_dsxx::UnitTest &ut) {
+
+  // Make true if golds need updating
+  const bool do_print = false;
+
+  // Tolerance used for checks
+  const double tol = 1e-11;
+
+  // Start the test.
+
+  std::cout << "\n---------------------------------------------------------\n"
+            << "   Test Draco code calling CSK_generator routines\n"
+            << "---------------------------------------------------------\n";
+
+  // open a small mg opacity file:
+  const std::string filename = ut.getTestSourcePath() + "../../compton2/test/dummy_data";
+  std::cout << "Attempting to construct a Compton object...\n" << std::endl;
+  std::unique_ptr<rtt_compton::Compton> compton_test;
+
+  try {
+    compton_test.reset(new rtt_compton::Compton(filename));
+  } catch (int /*asrt*/) {
+    FAILMSG("Failed to construct a Compton object!");
+    // if construction fails, there is no reason to continue testing...
+    return;
+  }
```

Comment:
Can do. This is a copy-paste from `src/compton/test/tCompton.cc`

Reply:
> Can do. This is a copy-paste from `src/compton/test/tCompton.cc`

Again, this can be left here if you are okay with the test aborting immediately if some other exception is thrown.

Reply:
Yes, that's fine with me